pkg/controller/networkaddonsconfig/networkaddonsconfig_controller_test.go

```diff
@@ -0,0 +1,166 @@
+package networkaddonsconfig
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/kubevirt/cluster-network-addons-operator/pkg/names"
+	"github.com/kubevirt/cluster-network-addons-operator/pkg/render"
+)
+
+type checkUnit struct {
+	key           string
+	shouldExist   bool
+	expectedValue string
+}
+
+var _ = Describe("Networkaddonsconfig", func() {
+	const testDataFile = "testdata/data.yaml"
+	const numOfObjectsInDataFile = 4
+
+	Context("When CR is not labeled", func() {
```

Comment:
Can we change the unit test to a table, where `crLabels` and the expected `[]checkUnit` are the input?

Reply:
I think it will be problmematic, because as we can see current implementation each of the tests / BeforeEach creates the specific crLabels and checkUnit it needs, wrapped by the right context 

creating a table means there will be a mixed section that will create them all

Reply:
> I think it will be problmematic, because as we can see current implementation each of the tests / BeforeEach creates the specific crLabels and checkUnit it needs, wrapped by the right context

if you move the `updateObjectsLabels()` to the test body, then you can use the input crLabels&checkUnit in it with no issue

>creating a table means there will be a mixed section that will create them all,
context will be lost, as it wraps nicely the current context imo atm
for example when crLables is not tagged, and when crLables is tagged is separated

I see what you mean, I usually convey the context inside each table entry, like so
```
Entry("When CR is not labeled, Should find only default relationship labels", 
updateObjectsLabelsParams{			
expectedCheckUnit: {...},
crLabels: {...}
}),
```
This way the context is not lost. another example [here](https://github.com/kubevirt/cluster-network-addons-operator/blob/master/tools/bumper/cnao_repo_commands_test.go#L131)

Reply:
I understand, thanks, but I find it much more readable as its now,
I would prefer to keep it this way please if possible,
I think the suggested method will be too dense for those tests, as they have parameters and logic that is more readable at the current state.

Moreover as suggested by you before and done, `updateObjectsLabels` more natural place is the current one in the `BeforeEach`

Reply:
I find it neater, but I won't insist on it.